### PR TITLE
add documentation for new aggregate command + auto aggregation

### PR DESCRIPTION
**Pull request by OVI3D0**
### Description
Adds documentation for the new `aggregate` command, introduced in OSB [here](https://github.com/opensearch-project/opensearch-benchmark/pull/638), as well as automatic aggregation of test results, introduced [here](https://github.com/opensearch-project/opensearch-benchmark/pull/655).

### Version
OpenSearch Benchmark

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _benchmark/reference/commands/aggregate.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+nav_order: 85
+parent: Command reference
+grand_parent: OpenSearch Benchmark Reference
+redirect_from: 
+  - /benchmark/commands/aggregate/
+---
+
+# Aggregation
+
+OpenSearch Benchmark now supports result aggregation, allowing users to combine multiple test executions into a single aggregated result. This feature enhances the benchmarking workflow by providing a more streamlined way to conduct and analyze multiple test runs.
```
Comment (Naarcha-AWS) [inline _benchmark/reference/commands/aggregate.md:13]:
```suggestion
Result aggregation allows users to combine multiple test executions into a single aggregated result, providing a more streamlined way to conduct and analyze multiple test runs.
```

**Review by Naarcha-AWS — APPROVED** [review]

File: _benchmark/reference/commands/aggregate.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+redirect_from: 
+  - /benchmark/commands/aggregate/
+---
+
+# aggregate
+
+The `aggregate` command combines multiple test executions into a single aggregated result, providing a more streamlined way to conduct and analyze multiple test runs. There are two methods of aggregation:
+
+- [Auto-aggregation](#auto-aggregation)
+- [Manual aggregation(#manual-aggregation)
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:16]:
```suggestion
- [Manual aggregation](#manual-aggregation)
```

File: _benchmark/reference/commands/aggregate.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Auto-aggregation
+
+Auto-aggregation runs multiple iterations of benchmark tests and automatically aggregate the results, all within a single command. You can use the flags outlined in this with the `execute` command.
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:20]:
```suggestion
The auto-aggregation method runs multiple iterations of benchmark tests and automatically aggregates the results, all within a single command. You can use the flags outlined in this with the `execute` command.
```

File: _benchmark/reference/commands/aggregate.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:21]: Above: It looks like there's a missing word in the second sentence. This what?

File: _benchmark/reference/commands/aggregate.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+The following example runs the `geonames` workload and aggregates the results twice: 
+
+```bash
+opensearch-benchmark execute --test-iterations=2 --aggregate=true --workload=geonames --target-hosts=127.0.0.1:9200
+```
+{% include copy-curl.html %}
+
+### Auto-aggregation flags
+
+The following new flags can be used to customized auto-aggregation:
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:33]:
```suggestion
The following new flags can be used to customize the auto-aggregation method:
```

File: _benchmark/reference/commands/aggregate.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+- `--test-iterations`: Specifies the number of times to run the workload (default: 1).
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:35]:
```suggestion
- `--test-iterations`: Specifies the number of times to run the workload (default is `1`).
```

File: _benchmark/reference/commands/aggregate.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+- `--aggregate`: Determines whether to aggregate the results of multiple test executions (default: true).
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:36]:
```suggestion
- `--aggregate`: Determines whether to aggregate the results of multiple test executions (default is `true`).
```

File: _benchmark/reference/commands/aggregate.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+- `--sleep-timer`: Specifies the number of seconds to sleep before starting the next test execution (default: 5).
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:37]:
```suggestion
- `--sleep-timer`: Specifies the number of seconds to sleep before starting the next test execution (default is `5`).
```

File: _benchmark/reference/commands/aggregate.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+- `--cancel-on-error`: When set, stops executing tests if an error occurs in one of the test iterations (default: false).
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:38]:
```suggestion
- `--cancel-on-error`: When set, stops executing tests if an error occurs in one of the test iterations (default is `false`).
```

File: _benchmark/reference/commands/aggregate.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+## Manual aggregation
+
+Use the `aggregate` command to manually aggregate results from multiple test executions:
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:42]:
```suggestion
You can use the `aggregate` command to manually aggregate results from multiple test executions.
```

File: _benchmark/reference/commands/aggregate.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+### Usage
+
+To aggregate multiple test executions manually, you can use the aggregate command:
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:46]:
```suggestion
To aggregate multiple test executions manually, you can use the `aggregate` command:
```

File: _benchmark/reference/commands/aggregate.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:47]: Above: This sentence is a rehash of the one under the preceding heading.

File: _benchmark/reference/commands/aggregate.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+```bash
+opensearch-benchmark aggregate --test-executions=<test_execution_id1>,<test_execution_id2>,...
+```
+{% include copy-curl.html %}
+
+### Response
+
+OpenSearch Benchmark responds with the following results:
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:55]:
```suggestion
OpenSearch Benchmark responds with the following:
```

File: _benchmark/reference/commands/aggregate.md [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+    /_/
+
+Aggregate test execution ID:  aggregate_results_geonames_9aafcfb8-d3b7-4583-864e-4598b5886c4f
+
+-------------------------------
+[INFO] SUCCESS (took 1 seconds)
+-------------------------------
+```
+
+The results will be aggregated into one test execution and stored under the ID shown in the output.
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:72]:
```suggestion
The results will be aggregated into one test execution and stored under the ID shown in the output:
```

File: _benchmark/reference/commands/aggregate.md [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+
+
+- `--test-execution-id`: Define a unique ID for the aggregated test execution.
+- `--results-file`: Write the aggregated results to the provided file.
+- `--workload-repository`: Define the repository from where OpenSearch Benchmark will load workloads (default: default).
```
Comment (natebower) [inline _benchmark/reference/commands/aggregate.md:78]:
```suggestion
- `--workload-repository`: Define the repository from which OpenSearch Benchmark will load workloads (default is `default`).
```

**Review by natebower — COMMENTED** [review]
@OVI3D0 @Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**